extensions/stackdriver/stackdriver.h

```diff
@@ -155,6 +162,8 @@ class StackdriverContext : public Context {
 
     is_tcp_ = true;
     getRootContext()->addToTCPRequestQueue(context_id_);
+    getRootContext()->setConnectionState(
```

Comment:
I think you will need to trigger log record in connection open, otherwise it will not be recorded. Also I think it is necessary to record log entry in open and close event, so that they could have correct time stamp.

Reply:
And test make sure that all three types of connection states are covered.

Reply:
Yes.. I was going to ping you offline about this :) I had exact same question in my other PR.. https://github.com/istio/proxy/pull/2843#discussion_r424828627 but reading it now, I think you were referring to Metrics and I was referring to calling Record method in general that does everything.

One of the things for TCP would be that metadata exchange doesn't happen until data comes so, connection open log won't have peer data. But it would be good to still get connection open logged.

Reply:
Will add this change in a separate PR because of metadata exchange stuff..

Reply:
Ok, I think time stamp is the only thing special with connection open. All other labels should have no difference from other state. You can keep an connection open time stamp somewhere and use that as a signal to record connection open event.

Reply:
Are you planning to fix this in separated PR?

Reply:
Yes, will do in a separate PR.. started a conversation about this with you, Mandar and Doug..